apis/kueue/v1beta1/localqueue_types.go

```diff
@@ -73,6 +73,10 @@ type LocalQueueFlavorStatus struct {
 	// +kubebuilder:validation:MaxItems=8
 	// +optional
 	NodeTaints []corev1.Taint `json:"nodeTaints,omitempty"`
+
+	// topologyName is the name of the topology associated with this ResourceFlavor.
```

Comment:
This seems to be mixing alpha and beta apis, no?

Reply:
We already used TopologyReference on ResourceFlavor:

https://github.com/kubernetes-sigs/kueue/blob/2bea78f84cd7430fcb104f07d447148094a4afd9/apis/kueue/v1beta1/resourceflavor_types.go#L98-L103

Reply:
:crying_cat_face: 

So we are planning on dropping alpha APIs from Kueue in our Kueue managed project. We were thinking that we would drop TAS CRDs and not set the feature gate.

What would happen if we add these references in the beta APIs? Are these no-opts since TAS is not defined?

Reply:
> 😿
> 
> So we are planning on dropping alpha APIs from Kueue in our Kueue managed project. We were thinking that we would drop TAS CRDs and not set the feature gate.
> 
> What would happen if we add these references in the beta APIs? Are these no-opts since TAS is not defined?

Not sure I understand the issue. We will move TAS CRDs to v1beta1 - nothing changes. Is there another way to link ResourceFlavor with the v1alpha1 Topology resource?

Reply:
I am trying to understand if one does not install the TAS CRDs, will there still be a dependency on resourceflavors/LocalQueues to an alpha feature?

Reply:
We don't reference the alpha API in any strong way. The "TopologyReference" is just an alias to string. If you delete Topology CRD the system will continue to run.
Sometimes, we need to extend more mature API (like LQ or RF) with a field referencing alpha API. The same happens for the core k8s. For example, the Pod's ResourceRequirements is extended with alpha (now beta) ResourceClaims (see [here](https://github.com/kubernetes/kubernetes/blob/b82260f00370bb77814d9c1f0a88d7a5724984bc/pkg/apis/core/types.go#L2497)). What we could do is add a comment on the field to say this is an "// +featureGate=TopologyAwareScheduling", analogously as done for ResourceClaim? wdyt @kannon92 ?

Reply:
I don't think we'll encounter any problems. If the feature gate is disabled, this field should be nil.

Reply:
> What we could add a comment on the field to say this is an "// +featureGate=TopologyAwareScheduling", analogously as done for ResourceClaim?

This is a good point.

Reply:
I'm happy with a feature gate flag but I think there may be code in Kueue required to handle that marker?

Reply:
I don't really know what +featureGate=TopologyAwareScheduling does actually..

Reply:
> I'm happy with a feature gate flag but I think there may be code in Kueue required to handle that marker?

Not based on LQ, this is purely informational API field at this point.

Reply:
> I don't really know what +featureGate=TopologyAwareScheduling does actually..

We can add more details like:
```
// topologyName is the name of the Topology referenced by the ResourceFlavor.
// The field is populated when the TopologyAwareScheduling feature gate is enabled.
```